hail/python/hailtop/aiotools/fs.py

```diff
@@ -522,10 +522,6 @@ def add_source_reports(transfer_report):
             print(f'  {sr._source}: {sr._files} files, {humanize.naturalsize(sr._bytes)}')
 
 
-class UnexpectedEOFError(Exception):
```

Comment:
This is going to cause a conflict when you rebase on main because of commit 89abe08339 which changes `_copy_part` to correctly throw UnexpectedEOFError when it does not read the entire part.

Let's put the exceptions into `aiotools/exceptions.py`.